File: src/modules/algorithm/seed/algorithms/binary-tree-lowest-common-ancestor/python/exercise.py
from typing import Optional

class TreeNode:
    def __init__(self, val: int = 0, left: Optional['TreeNode']=None, right: Optional['TreeNode']=None):
        self.val = val
        self.left = left
        self.right = right

def lowest_common_ancestor(root: Optional[TreeNode], p: TreeNode, q: TreeNode) -> Optional[TreeNode]:
    """
    Finds the lowest common ancestor of two nodes in a binary tree.
    
    Args:
        root: The root of the binary tree.
        p: The first node.
        q: The second node.
    
    Returns:
        The lowest common ancestor node, or None if not found.
    """
    # TODO: Implement the LCA algorithm
    return None 

File: src/modules/algorithm/seed/algorithms/binary-tree-lowest-common-ancestor/typescript/exercise.ts
export class TreeNode {
  val: number;
  left: TreeNode | null;
  right: TreeNode | null;

  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
    this.val = val === undefined ? 0 : val;
    this.left = left === undefined ? null : left;
    this.right = right === undefined ? null : right;
  }
}

/**
 * Finds the lowest common ancestor of two nodes in a binary tree.
 * @param root - The root of the binary tree.
 * @param p - The first node.
 * @param q - The second node.
 * @returns The lowest common ancestor node, or null if not found.
 */
export function lowestCommonAncestor(
  root: TreeNode | null,
  p: TreeNode,
  q: TreeNode,
): TreeNode | null {
  // TODO: Implement the LCA algorithm
  return null;
}
